moped-editor/src/views/projects/projectView/ProjectComponents.js

```diff
@@ -185,124 +187,109 @@ const ProjectComponents = () => {
+            projectFeatureCollection={projectFeatureCollection}
+            setIsEditing={false}
+            noPadding
+          >
+            <Button
+              className={classes.componentButtonAddNew}
+              onClick={handleAddNewComponentClick}
+              variant="outlined"
+              color="default"
+              size={"large"}
+              startIcon={<Icon>add</Icon>}
+              fullWidth
+            >
+              Add new component
+            </Button>
             {componentsAvailable && (
-              <Grid item xs={12} md={6}>
-                <ErrorBoundary
-                  FallbackComponent={({ error, resetErrorBoundary }) => (
-                    <ProjectSummaryMapFallback
-                      error={error}
-                      resetErrorBoundary={resetErrorBoundary}
-                      projectId={projectId}
-                      setIsEditing={null}
-                      refetchProjectDetails={refetch}
-                      mapData={projectFeatureCollection}
-                    />
-                  )}
-                  onReset={() => setMapError(false)}
-                  resetKeys={[mapError]}
-                >
-                  <ProjectComponentsMapView
-                    projectFeatureCollection={projectFeatureCollection}
-                    setIsEditing={false}
-                  />
-                </ErrorBoundary>
-              </Grid>
+              <ClickAwayListener onClickAway={handleComponentClickAway}>
+                <List className={classes.root}>
+                  {data.moped_proj_components.map((component, compIndex) => {
+                    const projComponentId = component.project_component_id;
+                    return (
+                      <ListItem
+                        key={"mcListItem-" + projComponentId}
+                        role={undefined}
+                        dense
+                        button
+                        onClick={() => handleComponentClick(projComponentId)}
+                        className={
+                          projComponentId === selectedComp
+                            ? classes.componentItemBlue
+                            : classes.componentItem
+                        }
+                      >
+                        <ListItemIcon>
+                          <Checkbox
+                            edge="start"
+                            className={classes.listItemCheckbox}
+                            checked={projComponentId === selectedComp}
+                            tabIndex={-1}
+                            disableRipple
+                            inputProps={{ "aria-labelledby": null }}
+                          />
+                        </ListItemIcon>
+                        <ListItemText
+                          id={"mcListItemText-" + projComponentId}
+                          primary={component?.moped_components?.component_type}
+                          secondary={
```

Comment:
this is a lot of logic to put within a prop. I wonder if the code would be neater if you could define this output as a variable and move into a separate function. Could also allow for more comment explaining the intention.